docs/guides/modules/hashlib.rst

```diff
@@ -0,0 +1,21 @@
+Hashlib
+=======
+A client module that provides several hashing algorithms.
+
+Usage
+-----
+The module provides the functions ``sha1``, ``sha256``, ``sha384`` and ``sha512``. Each
+can be called by passing the data to be hashed and will return a hex string.
+
+The data can be any object. If it is anything other than a string or bytes instance, it
+must have ``__str__`` method available.
```

Comment:
when you say it must have a `__str__` - all python objects inherit a `__str__` so not sure what the sentence is trying to add?
might be better to say it will be converted to a `str` object before being hashed.

But i'd still consider a `TypeError` for anything other than `str` or `bytes`.